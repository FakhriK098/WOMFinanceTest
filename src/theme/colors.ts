export const colors = {
  primary: '#4F46E5',
  danger: '#DC2626',
  white: '#FFFFFF',
  black: '#000000',
  text: {
    light: '#111827',
    dark: '#F9FAFB',
    mutedLight: '#6B7280',
    mutedDark: '#9CA3AF',
  },
  background: {
    light: '#FFFFFF',
    dark: '#0B1220',
  },
  border: {
    light: '#E5E7EB',
    dark: '#374151',
  },
  inputBg: {
    light: '#1118270D',
    dark: '#111827',
  },
  overlay: 'rgba(0,0,0,0.2)',
}
