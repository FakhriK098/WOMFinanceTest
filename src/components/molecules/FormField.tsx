import React from 'react';
import { View, StyleSheet } from 'react-native';
import Text from '../atoms/Text';
import Input from '../atoms/Input';
import type { FormFieldProps } from '@typings/components';
import { colors } from '@theme/colors';
import { typography } from '@theme/typography';
import Spacer from '../atoms/Spacer';
import { spacing } from '@theme/spacing';

const FormField = ({
  label,
  placeholder,
  value,
  onChangeText,
  secureTextEntry,
  errorText,
  keyboardType = 'default',
}: FormFieldProps) => {
  return (
    <View style={styles.container}>
      <Text size={typography.sm} color={colors.text.mutedLight}>
        {label}
      </Text>
      <Spacer size={spacing.sm} />
      <Input
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        secureTextEntry={secureTextEntry}
        keyboardType={keyboardType}
        error={Boolean(errorText)}
      />
      {!!errorText && (
        <Text
          size={typography.xs}
          color={colors.danger}
          style={styles.textError}
        >
          {errorText}
        </Text>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  textError: {
    marginTop: 6,
  },
});

export default React.memo(FormField);
