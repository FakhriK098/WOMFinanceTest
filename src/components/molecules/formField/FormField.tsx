import React from 'react';
import { View, StyleSheet } from 'react-native';
import Text from '@components/atoms/text/Text';
import Input from '@components/atoms/input/Input';
import type { FormFieldProps } from '@typings/components';
import { colors } from '@theme/colors';
import { typography } from '@theme/typography';
import Spacer from '@components/atoms/spacer/Spacer';
import { spacing } from '@theme/spacing';

const FormField = ({
  label,
  placeholder,
  value,
  onChangeText,
  secureTextEntry,
  errorText,
  keyboardType = 'default',
}: FormFieldProps) => {
  return (
    <View style={styles.container}>
      <Text size={typography.sm} color={colors.text.mutedLight}>
        {label}
      </Text>
      <Spacer size={spacing.sm} />
      <Input
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        secureTextEntry={secureTextEntry}
        keyboardType={keyboardType}
        error={Boolean(errorText)}
      />
      {!!errorText && (
        <Text
          size={typography.xs}
          color={colors.danger}
          style={styles.textError}
        >
          {errorText}
        </Text>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  textError: {
    marginTop: 6,
  },
});

export default React.memo(FormField);
