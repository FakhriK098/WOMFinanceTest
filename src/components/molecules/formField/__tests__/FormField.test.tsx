import React from 'react';
import renderer, { act } from 'react-test-renderer';

import FormField from '../FormField';
import { colors } from '@theme/colors';

describe('FormField molecule', () => {
  it('matches snapshot (no error)', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <FormField
          label="Email"
          placeholder="Your email"
          value=""
          onChangeText={() => {}}
        />,
      );
    });
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });

  it('renders error text and sets Input error', () => {
    let tr: renderer.ReactTestRenderer;
    act(() => {
      tr = renderer.create(
        <FormField
          label="Password"
          placeholder="Password"
          value=""
          onChangeText={() => {}}
          secureTextEntry
          errorText="Required"
        />,
      );
    });
    const errorTextNode = tr!.root.find(
      (inst) => inst.props && inst.props.children === 'Required',
    );
    expect(errorTextNode).toBeTruthy();
    // Verify that some wrapper within Input applies danger border when error is true
    const styledNodes = tr!.root.findAll((inst) => inst.props && inst.props.style);
    const hasDanger = styledNodes.some((inst) => {
      const style = inst.props.style as any;
      const arr = Array.isArray(style) ? style : [style];
      return arr.some((s) => s && s.borderColor === colors.danger);
    });
    expect(hasDanger).toBe(true);
    expect(tr!.toJSON()).toMatchSnapshot();
    act(() => tr!.unmount());
  });
});
