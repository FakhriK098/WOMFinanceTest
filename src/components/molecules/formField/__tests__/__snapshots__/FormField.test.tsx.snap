// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField molecule matches snapshot (no error) 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#6B7280",
          "fontSize": 14,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Email
  </Text>
  <View
    style={
      {
        "height": 8,
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 12,
          "borderWidth": 1,
          "paddingHorizontal": 14,
        },
        {
          "backgroundColor": "#1118270D",
          "borderColor": "#E5E7EB",
        },
        false,
        false,
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Your email"
      placeholderTextColor="#6B7280"
      style={
        [
          {
            "height": 48,
          },
          {
            "color": "#111827",
          },
          undefined,
        ]
      }
      value=""
    />
  </View>
</View>
`;

exports[`FormField molecule renders error text and sets Input error 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#6B7280",
          "fontSize": 14,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    Password
  </Text>
  <View
    style={
      {
        "height": 8,
        "width": 8,
      }
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 12,
          "borderWidth": 1,
          "paddingHorizontal": 14,
        },
        {
          "backgroundColor": "#1118270D",
          "borderColor": "#E5E7EB",
        },
        false,
        {
          "borderColor": "#DC2626",
        },
      ]
    }
  >
    <TextInput
      autoCapitalize="none"
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      placeholderTextColor="#6B7280"
      secureTextEntry={true}
      style={
        [
          {
            "height": 48,
          },
          {
            "color": "#111827",
          },
          undefined,
        ]
      }
      value=""
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#DC2626",
          "fontSize": 12,
          "fontWeight": "400",
        },
        {
          "marginTop": 6,
        },
      ]
    }
  >
    Required
  </Text>
</View>
`;
